examples/src/main/spark/org/apache/ignite/examples/spark/IgniteDataFrameJoinExample.scala

```diff
@@ -0,0 +1,176 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.examples.spark
+
+import java.lang.{Integer => JInt, Long => JLong, String => JString}
+
+import org.apache.ignite.cache.query.SqlFieldsQuery
+import org.apache.ignite.configuration.CacheConfiguration
+import org.apache.ignite.spark.IgniteDataFrameSettings._
+import org.apache.ignite.{Ignite, Ignition}
+import org.apache.log4j.{Level, Logger}
+import org.apache.spark.sql.SparkSession
+
+/**
+  * Example application demonstrates the join operations between two dataframes or Spark tables with data saved in Ignite caches.
+  */
+object IgniteDataFrameJoinExample extends App {
+    /**
+      * Ignite config file.
+      */
+    private val CONFIG = "examples/config/example-ignite.xml"
+
+    /**
+      * Test cache name.
+      */
+    private val CACHE_NAME = "testCache"
+
+    //Starting Ignite server node.
+    val ignite = setupServerAndData
+
+    closeAfter(ignite) { ignite ⇒
+        //Creating spark session.
+        implicit val spark = SparkSession.builder()
+            .appName("IgniteDataFrameJoinExample")
+            .master("local")
+            .config("spark.executor.instances", "2")
+            .getOrCreate()
+
+        // Adjust the logger to exclude the logs of no interest.
+        Logger.getRootLogger.setLevel(Level.ERROR)
+        Logger.getLogger("org.apache.ignite").setLevel(Level.INFO)
+
+        // Executing examples.
+
+        sparkDSLJoinExample
+        nativeSparkSqlJoinExample
+    }
+
+    /**
+      * Examples of usage Ignite DataFrame implementation.
+      * Selecting data throw Spark DSL.
+      *
+      * @param spark SparkSession.
+      */
+    def sparkDSLJoinExample(implicit spark: SparkSession): Unit = {
+        println("Querying using Spark DSL.")
+        println
+
+        val persons = spark.read
+          .format(FORMAT_IGNITE)
+          .option(OPTION_TABLE, "person")
+          .option(OPTION_CONFIG_FILE, CONFIG)
+          .load()
+
+        persons.printSchema()
+        persons.show()
+
+        val cities = spark.read
+          .format(FORMAT_IGNITE)
+          .option(OPTION_TABLE, "city")
+          .option(OPTION_CONFIG_FILE, CONFIG)
+          .load()
+
+        persons.printSchema()
+        persons.show()
+
+        val joinResult = persons.join(cities, persons("city_id") === cities("id"))
+          .select(persons("name").as("person"), persons("age"), cities("name").as("city"), cities("country"))
+
+        joinResult.explain(true)
+        joinResult.printSchema()
+        joinResult.show()
+    }
+
+    /**
+      * Examples of usage Ignite DataFrame implementation.
```

Comment:
Javadoc not formatted properly.

Reply:
Is Ignite JavaDoc formatting applied to Scala code?